lib/slack-ruby-bot/hooks/hook_support.rb

```diff
@@ -0,0 +1,39 @@
+module SlackRubyBot
+  module Hooks
+    module HookSupport
+      def self.included(base)
+        base.cattr_accessor :hook_blocks
+
+        base.extend(ClassMethods)
+      end
+
+      module ClassMethods
+        def on(event_name, &block)
+          self.hook_blocks ||= {}
+
+          self.hook_blocks[event_name] ||= []
+          self.hook_blocks[event_name] << block
+        end
+      end
+
+      # Instance stuff
+      attr_writer :hooks
```

Comment:
sorry, you're correct. This was a previous iteration, I don't need either as I'm implementing a reader method anyway. 

of with it's head!